DRTM with TrenchBoot - introduction (#217)
3mdeb/news-and-ideas: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md

---
title: Open Source DRTM with TrenchBoot for AMD processors. Introduction.
abstract: This article starts an entire series of articles related to title
          project. Today, you will find out why we have started such project and
          who is supporting us. Also, we bring you closer to main concept and
          goals.
cover: /covers/nlnet-logo.png
author: piotr.kleinschmidt
layout: post
published: false
date: 2020-03-24
archives: "2020"

tags:
  - trenchboot
  - security
  - open-source
  - coreboot
categories:
  - Firmware
  - Security

---

## Introduction

> Our company activity is to provide secure firmware and embedded systems to our

Author: Done

customers. Our strategy is to use, contribute and spread open source solutions
as much as possible. Factor, motivating to further work and showing the
legitimacy of the adopted philosophy, is feedback in the form of satisfied

> customers, community and incoming foundations.

Author: customers and community. We also greatly appreciate the help from foundations which want to support our projects and ideas.
@pietrushnic I suggest such change. Is it ok?

> customers, community and incoming foundations.

Author: Done

We are glad to inform, that we have received a grant from  **NLnet Foundation**
for *Open Source DRTM implementation with TrenchBoot for AMD processors*
project. Under the subsidy, **3mdeb Embedded Systems Consulting** wants to
contribute these solutions to the public good, not as proprietary products, but
as free and open source technologies that any person or application can use
without restrictions and modify to their own needs and the needs of others.

This article is an introduction, which should bring you closer to the overall
project's concept and describe goals, which we want to achieve. Also, this blog
post starts a series of articles describing and verifying our progress at work.
Thanks to that, you can observe development process and always be up-to-date
about the project.

## Our goal

Our goal is twofold:
1. Implement DRTM with TrenchBoot for AMD processors.

> 2. Create test environment with test suites for users.

Author: Changed

Above issues will be described later. Both, as already mentioned, will be fully

> open-source and available for everybody. Solution, which we provide will be

Author: Done

intended for AMD processors because it doesn't include any closed

> components.

Reviewer: @krystian-hebel I would add that there 2 ACMs, ACM SINIT which can be downloaded from Intel website but is not redistributable, and ACM BIOS, which can be obtained only as OEM under CNDA with Intel.

> components.

Reviewer 2: I wasn't sure about that so decided to write it in a safe way :) Thanks for clarification

> components.

Author: I mixed both comments to a one description. Please verify if it is okay.

Now, let me briefly discuss each goal, so you can better understand the project.
Of course, we won't focus on details and all requirements now. Those will be
systematically updated and presented to community.

#### DRTM with TrenchBoot

> TrenchBoot is one of the stages of boot process. It runs directly after firmware

Author: @krystian-hebel  As it is introductory blog post, I don't delve into TrenchBoot structure too much. Let's leave technical details for future articles.

(coreboot in our case) execution ended. It is a technology which measures and
verifies running environment. To obtain that, TrenchBoot utilizes **Dynamic Root
of Trust for Measurements** (DRTM). It is very complex topic and understanding

> it could take too much time. What you need to know (or trust us about) is that

Author: Done

those measurements are done for given piece of code (given piece of system) and
are stored in **Trusted Platform Module** (TPM) special registers. It gives
possibility to automatically verify them against corruption and eventually

> proceed or stop further booting. Also, the user has possibility to check if

Author: Done

> measurements are valid. It can be done from OS level using dedicated tools. So,

Author: Done

> as you can see, TrenchBoot is critical piece of firmware, if you want to make it

Author: Done

secure. Also, its implementation is fully open-source, so everyone can inspect
it and contribute.

#### Test environment

> All requirements of project are automatically or manually validate after

Author: Done

development. For this purpose, we need to create entire testing infrastructure
with suitable hardware and software components. Moreover, all tests will be
published for users, so they can verify development stage on their own
platforms.

Final results, most likely at the end of each month, will be presented in form
of blog articles. Everything should be clearly pointed out and explained. All
used tools, all carried out procedures, all prepared configurations - everything
user can inspect, revise, build, reproduce and finally use. The goal is to have
a user-friendly and trust-worthy environment, so community can be sure about the
quality of the solution we provide.

> ## Summary

Reviewer: @piotr-kleins we have to mention main TrenchBoot repository and other individuals and companies that contribute to the project Oracle, Daniel Smith, Ross, Daniel Kiper, and others. @krystian-hebel cc

Reviewer 2: List of all commiters/reviewers of LZ only:
```
Andrew Cooper
Daniel P. Smith
Krystian Hebel
Michał Żygowski
Piotr Król
Ross Philipson
```
In addition to those there is also one commit from Eric Snowberg to kernel, and of course Daniel Kiper reviews patches send to GRUB.

Author: Added in summary section

As I mentioned in introduction section, above article is just the beginning of
the entire series about **DRTM with TrenchBoot implementation**. So stay tuned
and wait for other articles!

Once again, we would like to thank **NLnet Foundation** for their support. We
believe that our project will benefit the open-source community and will be a
big step in development of firmware security.

If you think we can help in improving the security of your firmware or you
looking for someone who can boost your product by leveraging advanced features

> of used hardware platform, feel free to [book a call with

Reviewer 2: Does it render properly with line break here?

Author: Revert to original formatting

---
title: Open Source DRTM with TrenchBoot for AMD processors. Introduction.
abstract: This article starts an entire series of articles related to title

>           project. Today, you will find out why we have started such project and

Reviewer 3: As in other post I suggest to not use time-related words. Please use something like: `by reading this blog post, you will find out ...`

Author: done

          who is supporting us. Also, we bring you closer to main concept and
          goals.
cover: /covers/nlnet-logo.png
author: piotr.kleinschmidt
layout: post
published: false
date: 2020-03-26
archives: "2020"

tags:
  - trenchboot
  - security
  - open-source
  - coreboot
categories:
  - Firmware
  - Security

---

## Introduction

Our company's activity is to provide secure firmware and embedded systems to our

> customers. Our strategy is to use, contribute and spread open source solutions

Reviewer 3: ```suggestion
customers. 3mdeb strategy is to use, contribute and spread open source solutions
```
Too many `our`

Author: done

project. Under the subsidy, **3mdeb Embedded Systems Consulting** wants to
contribute these solutions to the public good, not as proprietary products, but
as free and open source technologies that any person or application can use
without restrictions and modify to their own needs and the needs of others.

This article is an introduction, which should bring you closer to the overall
project's concept and describe goals, which we want to achieve. Also, this blog
post starts a series of articles describing and verifying our progress at work.
Thanks to that, you can observe development process and always be up-to-date
about the project.

## Our goal

Our goal is twofold:
1. Implement DRTM with TrenchBoot for AMD processors.
2. Create a test environment with test suites for the community.

Above issues will be described later. Both, as already mentioned, will be fully
open-source and available for everybody. Solution which we provide will be
intended for AMD processors because it doesn't include any closed software
components. To do something similar on Intel processors, components called ACM
(Authenticated Code Module) are required. More precisely, first is ACM SINIT

> which be downloaded from Intel website, but is not redistributable. Second is

Reviewer 3: ```suggestion
which can be downloaded from Intel website, but is not redistributable. Second is
```

Author: done

ACM BIOS which can be obtained only as OEM under CNDA with Intel. Hence, it
denies open-source idea. Also, ACMs are different for every CPU and must be
individually requested from Intel. Moreover, being delivered in a binary form,
there is no way to audit its code. In summary, there are many more problems with
Intel than with AMD.

Now, let me briefly discuss each goal, so you can better understand the project.
Of course, we won't focus on details and all requirements now. Those will be
systematically updated and presented to community.

#### DRTM with TrenchBoot

[TrenchBoot](https://github.com/TrenchBoot) *is a framework that allows
individuals and projects to build security engines to perform launch integrity

> actions for their systems.* In the other words, it gives a tools to create the

Reviewer 3: ```suggestion
actions for their systems.* In other words, it provides tools to create the
```

Author: Done
